fmriprep/workflows/bold/reference.py

```diff
@@ -0,0 +1,147 @@
+
+    This workflow assumes only one BOLD file has been passed.
+
+    Workflow Graph
+        .. workflow::
+            :graph2use: orig
+            :simple_form: yes
+
+            from fmriprep.workflows.bold.reference import init_raw_boldref_wf
+            wf = init_raw_boldref_wf()
+
+    Parameters
+    ----------
+    bold_file : :obj:`str`
+        BOLD series NIfTI file
+    multiecho : :obj:`bool`
+        If multiecho data was supplied, data from the first echo will be selected
+    name : :obj:`str`
+        Name of workflow (default: ``bold_reference_wf``)
+
+    Inputs
+    ------
+    bold_file : str
+        BOLD series NIfTI file
+    dummy_scans : int or None
+        Number of non-steady-state volumes specified by user at beginning of ``bold_file``
+
+    Outputs
+    -------
+    bold_file : str
+        Validated BOLD series NIfTI file
+    boldref : str
+        Reference image to which BOLD series is motion corrected
+    skip_vols : int
+        Number of non-steady-state volumes selected at beginning of ``bold_file``
+    algo_dummy_scans : int
+        Number of non-steady-state volumes agorithmically detected at
+        beginning of ``bold_file``
+
+    """
+    from niworkflows.interfaces.bold import NonsteadyStatesDetector
+    from niworkflows.interfaces.images import RobustAverage
+
+    workflow = Workflow(name=name)
+    workflow.__desc__ = f"""\
+First, a reference volume was generated{' from the shortest echo of the BOLD run' * multiecho},
+using a custom methodology of *fMRIPrep*, for use in head motion correction.
+"""
+
+    inputnode = pe.Node(
+        niu.IdentityInterface(fields=["bold_file", "dummy_scans"]),
+        name="inputnode",
+    )
+    outputnode = pe.Node(
+        niu.IdentityInterface(
+            fields=[
+                "bold_file",
+                "boldref",
+                "skip_vols",
+                "algo_dummy_scans",
+                "validation_report",
+            ]
+        ),
+        name="outputnode",
+    )
+
+    # Simplify manually setting input image
+    if bold_file is not None:
+        inputnode.inputs.bold_file = bold_file
+
+    val_bold = pe.Node(
+        ValidateImage(),
+        name="val_bold",
+        mem_gb=DEFAULT_MEMORY_MIN_GB,
+    )
+
+    get_dummy = pe.Node(NonsteadyStatesDetector(), name="get_dummy")
+    gen_avg = pe.Node(RobustAverage(), name="gen_avg", mem_gb=1)
+
+    calc_dummy_scans = pe.Node(
+        niu.Function(function=pass_dummy_scans, output_names=["skip_vols_num"]),
+        name="calc_dummy_scans",
+        run_without_submitting=True,
+        mem_gb=DEFAULT_MEMORY_MIN_GB,
+    )
+
+    # fmt: off
+    workflow.connect([
+        (inputnode, val_bold, [("bold_file", "in_file")]),
+        (inputnode, get_dummy, [("bold_file", "in_file")]),
+        (inputnode, calc_dummy_scans, [("dummy_scans", "dummy_scans")]),
+        (val_bold, gen_avg, [("out_file", "in_file")]),
+        (get_dummy, gen_avg, [("t_mask", "t_mask")]),
```

Comment:
Do we actually want to use dummy scans for the HMC reference? I think this is an improvement over sbref, but we should consider if we're still biasing the motion estimates too low.